frontend/tests/e2e/tests/groupmember/CancelGroupMemberTest.js

```diff
@@ -0,0 +1,68 @@
+import AlertPage from '../../pages/AlertPage';
+import { SITE_UNDER_TEST } from '../../configuration';
+import CancelGroupMember from '../../pages/groupmember/CancelGroupMember';
+import { regularAccUser } from '../../roles/roles';
+
+const ERROR_MESSAGE = 'Please correct errors before submitting'
+const PHN_REQUIRED_MESSAGE = 'PHN is required'
+const INVALID_PHN_ERROR_MESSAGE = 'PHN format is invalid'
+const GROUP_NUMBER_REQUIRED_MESSAGE = 'Group Number is required'
+const INVALID_GROUP_NUMBER_ERROR_MESSAGE = 'Group Number is invalid'
+const COVERAGE_CANCEL_DATE_REQUIRED_MESSAGE = 'Coverage Cancel Date is required'
+const CANCEL_REASON_REQUIRED_MESSAGE = 'Cancel Reason is required'
+
+const PAGE_TO_TEST = SITE_UNDER_TEST + '/groupmember/CancelGroupMember'
+
+fixture(`CancelGroupMember Page`)
+    .disablePageCaching `Test CancelGroupMember`
+    .beforeEach(async t => {
+        await t
+            .useRole(regularAccUser)
+    })
+    .page(PAGE_TO_TEST);
+
+test('Check required fields validation', async t => {
+    await t
+        // Given required fields aren't filled out (phn, group member, group member number, department)
+        // When I click the submit button
+        .click(CancelGroupMember.submitButton)
+        // I expect an error message stating the page had errors and individual error messages for each required field
+        .expect(AlertPage.alertBannerText.textContent).contains(ERROR_MESSAGE)
+        .expect(CancelGroupMember.errorText.nth(0).textContent).contains(GROUP_NUMBER_REQUIRED_MESSAGE)
+        .expect(CancelGroupMember.errorText.nth(1).textContent).contains(PHN_REQUIRED_MESSAGE)
+        .expect(CancelGroupMember.errorText.nth(2).textContent).contains(COVERAGE_CANCEL_DATE_REQUIRED_MESSAGE)
+        .expect(CancelGroupMember.errorText.nth(3).textContent).contains(CANCEL_REASON_REQUIRED_MESSAGE)
+});
+
+test('Check invalid formats validation', async t => {
+    await t
+        // Given I enter PHN, Group Number, and Cancel dates with invalid formats
+        .typeText(CancelGroupMember.phnInput, '9000448000')
+        .typeText(CancelGroupMember.groupNumberInput, '123')
+        // When I click the submit button
+        .click(CancelGroupMember.submitButton)
+        // I expect an error message stating the page had errors and an individual error message for the PHN and Group and Cancel Date formats
+        .expect(CancelGroupMember.errorText.nth(0).textContent).contains(INVALID_GROUP_NUMBER_ERROR_MESSAGE)
+        .expect(CancelGroupMember.errorText.nth(1).textContent).contains(INVALID_PHN_ERROR_MESSAGE)
+        .expect(AlertPage.alertBannerText.textContent).contains(ERROR_MESSAGE)
+});
+
+test('Check properly filled form passes validation', async t => {
+        // TODO (tschia) determine if we should test this since it will only pass once
```

Comment:
Probably still a good idea to have it as a record of the steps for a successful submit. Your call.

Reply:
test added